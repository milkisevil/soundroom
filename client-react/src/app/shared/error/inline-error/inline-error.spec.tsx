import * as React from 'react';
import InlineError from './inline-error';
import { create } from 'react-test-renderer';

const MOCK_CHILDREN = 'I am a child string';
const MOCK_MESSAGE = 'I am a custom message';
const MOCK_STATUS = 1234;

describe('<InlineError />', () => {
  it('renders children', () => {
    const inlineError = create(<InlineError>{MOCK_CHILDREN}</InlineError>).toJSON();
    expect(inlineError).toMatchSnapshot();
  });

  it('renders message prop', () => {
    const inlineError = create(<InlineError message={MOCK_MESSAGE}/>).toJSON();
    expect(inlineError).toMatchSnapshot();
  });

  it('renders status prop', () => {
    const inlineError = create(<InlineError status={MOCK_STATUS}/>);
    expect(inlineError).toMatchSnapshot();
  });

  it('renders both message and status prop', () => {
    const inlineError = create(<InlineError message={MOCK_MESSAGE} status={MOCK_STATUS}/>);
    expect(inlineError).toMatchSnapshot();
  });

  it('renders children, message and status prop', () => {
    const inlineError = create(<InlineError message={MOCK_MESSAGE} status={MOCK_STATUS}>{MOCK_CHILDREN}</InlineError>);
    expect(inlineError).toMatchSnapshot();
  });
});
