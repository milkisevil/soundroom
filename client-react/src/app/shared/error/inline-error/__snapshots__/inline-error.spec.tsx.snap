// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineError /> renders a basic 1`] = `
<div
  className="sc-bdVaJa fyJxkR"
>
   
</div>
`;

exports[`<InlineError /> renders both message and status prop 1`] = `
<div
  className="sc-bdVaJa fyJxkR"
>
   
  [status:1234 | I am a custom message]
</div>
`;

exports[`<InlineError /> renders children 1`] = `
<div
  className="sc-bdVaJa fyJxkR"
>
  I am a child string
   
</div>
`;

exports[`<InlineError /> renders children, message and status prop 1`] = `
<div
  className="sc-bdVaJa fyJxkR"
>
  I am a child string
   
  [status:1234 | I am a custom message]
</div>
`;

exports[`<InlineError /> renders message prop 1`] = `
<div
  className="sc-bdVaJa fyJxkR"
>
   
  [I am a custom message]
</div>
`;

exports[`<InlineError /> renders status prop 1`] = `
<div
  className="sc-bdVaJa fyJxkR"
>
   
  [status:1234]
</div>
`;
